### PR TITLE
chore: switch to using pnpm

### LINKED ISSUE #785: Switch to pnpm as package manager
### Is your enhancement related to a problem? Please describe

See https://github.com/containers/podman-desktop/issues/8566, 
https://github.com/containers/podman-desktop-extension-ai-lab/issues/1652.

### Describe the solution you'd like

Podman Desktop and other extensions are switching to pnpm, bootc should too.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Initial pass at moving to pnpm based on changes to other repos. All of the basic commands like build/watch work, but not 'npm run typecheck:frontend'.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Fixes #785.

### How to test this PR?

Execute normal development commands; PR build passes.

### CODE REVIEW

**Comment by benoitf** [comment]
I think that codebase should be fixed from sonarjs plug-in lint report (v2) then plug-in can be upgraded and it'll ease all the updates (pnpm, eslint, etc)

**Comment by benoitf** [comment]
I think order to have something successful is sonarjs, eslint v9 and then pnpm

**Comment by deboer-tim** [comment]
Yup, added a commit to #769.

File: package.json [hunk 1]
```diff
@@ -62,5 +62,6 @@
   },
   "dependencies": {
     "js-yaml": "^4.1.0"
-  }
+  },
+  "packageManager": "pnpm@9.7.1+sha512.faf344af2d6ca65c4c5c8c2224ea77a81a5e8859cbc4e06b1511ddce2f0151512431dd19e6aff31f2c6a8f5f2aced9bd2273e1fed7dd4de1868984059d2c4247"
```
Comment (benoitf) [inline package.json:66]: might bump to 9.10

**Review by benoitf — COMMENTED** [review]
might exclude yarn.lock file in the .gitignore file as well

**Comment by deboer-tim** [comment]
Bumped to 9.10 and added .gitignore for yarn.lock.

**Comment by benoitf** [comment]
https://github.com/containers/podman-desktop-extension-bootc/blob/4833fec8ac92cbfc0814e2a178375fd0ffcb4847/package.json#L57-L62 should be removed as well

**Comment by benoitf** [comment]
pr-check.yaml is still containing `yarn` reference

**Comment by benoitf** [comment]
there is a conflict else lgtm

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -150,14 +129,14 @@ jobs:
           echo "Version of @podman-desktop/tests-playwright to be used: $version"
           jq --arg version "$version" '.devDependencies."@podman-desktop/tests-playwright" = $version' package.json > package.json_tmp && mv package.json_tmp package.json
 
-      - name: Execute yarn in Bootc Extension
+      - name: Execute pnpm in Bootc Extension
         working-directory: ./podman-desktop-extension-bootc
-        run: yarn --frozen-lockfile
+        run: pnpm install
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:134]:
```suggestion
        run: pnpm install --no-frozen-lockfile
```
need to fetch a different version of test-playwright dependencies

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/main/.github/workflows/e2e-main.yaml#L117

Comment (deboer-tim) [reply .github/workflows/pr-check.yaml]: Fixed.

**Review by benoitf — APPROVED** [review]

**Pull request merged**